src/browser/modules/Stream/Stream.tsx

```diff
@@ -78,65 +80,70 @@ const getFrame = (type: any) => {
     dbs: DbsFrame,
     style: StyleFrame,
     edit: EditFrame,
-    default: Frame
+    default: DefaultFrame
   }
   return trans[type] || trans.default
 }
 
-class Stream extends PureComponent<any> {
-  base: any
-  componentDidMount() {
-    this.base = React.createRef()
-  }
+type StreamProps = {
+  frames: FrameStack[]
+  activeConnectionData: Connection | null
+  shouldScrollToTop: boolean
+}
 
-  componentDidUpdate(prevProps: any) {
+export interface BaseFrameProps {
+  frame: Frame & { isPinned: boolean }
+  activeConnectionData: Connection | null
+  stack: Frame[]
+}
+
+function Stream(props: StreamProps): JSX.Element {
+  const base = useRef<HTMLDivElement>(null)
+  const lastFrameCount = useRef(0)
+
+  useEffect(() => {
     // If we want to scroll to top when a new frame is added
     if (
-      prevProps.frames.length < this.props.frames.length &&
-      this.props.scrollToTop &&
-      this.base &&
-      this.base.current
+      lastFrameCount.current < props.frames.length &&
+      props.shouldScrollToTop &&
+      base.current
     ) {
-      this.base.current.scrollTop = 0
+      base.current.scrollTop = 0
     }
-  }
 
-  render() {
-    return (
-      <StyledStream ref={this.base} data-testid="stream">
-        {this.props.frames.map((frameObject: any) => {
-          const frame = getLatestFromFrameStack(frameObject)
-          const frameProps = {
-            frame: { ...frame, isPinned: frameObject.isPinned },
-            activeConnectionData: this.props.activeConnectionData,
-            stack: frameObject.stack
-          }
-          let MyFrame = getFrame(frame.type)
-          if (frame.type === 'error') {
-            try {
-              const cmd = frame.cmd.replace(/^:/, '')
-              const Frame = cmd[0].toUpperCase() + cmd.slice(1) + 'Frame'
-              MyFrame = require('./Extras/index')[Frame]
-              if (!MyFrame) {
-                MyFrame = getFrame(frame.type)
```

Comment:
This was written for us to sneak in fun frames without having to explicitly import them. I think we should keep it :)

Reply:
It's always easier to take fun out than put it in, I'll keep it as is then :)